cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerClientImpl.java

```diff
@@ -1571,17 +1571,15 @@ protected void doOpenFile(String urlPath, Object app, int instanceIndex, String
     protected void extractUriInfo(Map<String, UUID> domains, String uri, Map<String, String> uriInfo) {
         URI newUri = URI.create(uri);
         String host = newUri.getScheme() != null ? newUri.getHost() : newUri.getPath();
+
+        String matchedDomain = host.split("\\.", 2)[1];
         for (String domain : domains.keySet()) {
-            if (host != null && host.endsWith(domain)) {
-                String previousDomain = uriInfo.get("domainName");
-                if (previousDomain == null || domain.length() > previousDomain.length()) {
-                    // Favor most specific subdomains
-                    uriInfo.put("domainName", domain);
-                    if (domain.length() < host.length()) {
-                        uriInfo.put("host", host.substring(0, host.indexOf(domain) - 1));
-                    } else if (domain.length() == host.length()) {
-                        uriInfo.put("host", "");
-                    }
+            if (host != null && matchedDomain.equals(domain)) {
+                uriInfo.put("domainName", domain);
+                if (domain.length() < host.length()) {
```

Comment:
Obsolete code from the previous version. It was possible to have very specific domain that matches the concatanation of host + domain.
Now, when there is a check for host can be splitted in two parts with delimeter '.' it will not be possible